### ISSUE TITLE
thread 'main' panicked at src/tools/rustfmt/src/parse/session.rs:42:9: silent emitter attempted to translate a diagnostic

### ISSUE THREAD

**Issue by aDifferentJT** [rust-lang/rustfmt#6082]
[rustc-ice-2024-02-19T01_32_59-84679.txt](https://github.com/rust-lang/rustfmt/files/14325344/rustc-ice-2024-02-19T01_32_59-84679.txt)

Sorry, not a very helpful bug report, but it said to open an issue

**Comment by ytmimi** [comment]
No worries. What version of rustfmt are you using?

**Comment by aDifferentJT** [comment]
> No worries. What version of rustfmt are you using?

The latest nightly, I ran rustup update about an hour ago.

**Comment by aDifferentJT** [comment]
> @aDifferentJT Do you also have an input snippet that we could run rustfmt on to reproduce the issue?

Not to hand unfortunately, I haven't yet tried to narrow down the code that causes it, I'll try to do that in the morning, it's quite late here.

**Comment by ytmimi** [comment]
@aDifferentJT I don't think this is an issue with a particular code snippet. If I had to guess Id say changes made in https://github.com/rust-lang/rust/pull/121085 are causing this issue.

**Comment by ytmimi** [comment]
were you running rustfmt on a public repo? That might be enough to reproduce the issue.

**Comment by aDifferentJT** [comment]
> were you running rustfmt on a public repo? That might be enough to reproduce the issue.

Sorry, no, it's private for now, if I get time in the morning I'll try to get a more minimal case I can send

**Comment by aDifferentJT** [comment]
I've minimised the file, this seems to be the smallest file that causes the issue

[rustfmt_ice.rs](https://github.com/rust-lang/rustfmt/files/14332406/rustfmt_ice.rs.txt)

```
macro_rules! test {
    ($T:ident, $b:lifetime) => {
        Box<$T<$b>>
    };
}
```

**Comment by ytmimi** [comment]
Perfect! I'm also able to reproduce the bug using the snippet. Thank you very much! I think this will help us figure out what's going on.

**Comment by ytmimi** [comment]
https://github.com/rust-lang/rust/pull/121487 doesn't explicitly reference this issue, but I'm hoping that it also resolves the problem 🤞🏼

**Comment by ytmimi** [comment]
@aDifferentJT when you have a moment can you confirm that this is resolved on the latest nightly (rustfmt 1.7.0-nightly (7d3702e4 2024-03-06)? I just ran your input snippet with a few recent nighty versions, and I can no longer reproduce the panic using the most recent nightly.

**Comment by aDifferentJT** [comment]
@ytmimi Looks like it's working, thanks
